### ISSUE TITLE
evalf should always evaluate complex expressions in a FiniteSet to standard form

### ISSUE THREAD

**Issue by mhardcastle** [sympy/sympy#18281]
evalf should always evaluate complex expressions in a FiniteSet to standard form, shouldn't it?

```
In [1]:r=solveset(x**4+3*x**2+2*x+1)
In [2]: print(r.evalf())
FiniteSet(-1.74697789611327*I + 0.707106781186548*(-1 + 0.753397007321344*(-0.292306411107734 - I)**0.333333333333333 + 0.753397007321344*(-0.292306411107734 + I)**0.333333333333333)**0.5, -0.438990337475312*I - 0.707106781186548*(-1 + 0.753397007321344*(-0.292306411107734 - I)**0.333333333333333 + 0.753397007321344*(-0.292306411107734 + I)**0.333333333333333)**0.5, -0.707106781186548*(-1 + 0.753397007321344*(-0.292306411107734 - I)**0.333333333333333 + 0.753397007321344*(-0.292306411107734 + I)**0.333333333333333)**0.5 + 0.438990337475312*I, 0.707106781186548*(-1 + 0.753397007321344*(-0.292306411107734 - I)**0.333333333333333 + 0.753397007321344*(-0.292306411107734 + I)**0.333333333333333)**0.5 + 1.74697789611327*I)
```
But
```
In [3]: for v in r.args:
   print(v.evalf())
-0.349745826211722 - 0.438990337475312*I
0.349745826211722 - 1.74697789611327*I
-0.349745826211722 + 0.438990337475312*I
0.349745826211722 + 1.74697789611327*I
```

...

**Comment by gschintgen** [comment]
That's curious. Until a few days ago, `evalf` of a `FiniteSet` applied `._eval_evalf()` to its elements. But for symbolic elements that method is intended to return `None` which is of course not correct in this context. So I changed it to `._evalf(prec)` since that takes bits of precision as argument. (PR #18251)

But I must have mistaken `_evalf` for an internal equivalent of `evalf`, just with binary precision instead of decimal. At least that's what the docstring suggests:
https://github.com/sympy/sympy/blob/957f468b80325e33a40c37bbb50a6ca09e9784f8/sympy/core/evalf.py#L1474-L1475
But obviously it's not the case. So it seems that for this kind of usage the full regular `evalf` must be called, but with the precision converted from digits to bits. That's strange.

(Oh and somehow I'm getting a spurious intersection with `S.Complexes` instead of a plain `FiniteSet`. That must be another recent change.)

**Comment by mhardcastle** [comment]
Just for the record, I see this same behaviour both in the current dev version and in versions that are certainly more than a few days old.

**Comment by gschintgen** [comment]
Yes, you're right: In this respect `._eval_evalf` (older code) and `._evalf` (current code) behave similarly. What bothers me is that `._evalf` seemingly does not what it's documented to do.

**Comment by oscarbenjamin** [comment]
We should make a function `evalf` that also works for list/set/tuple.

**Comment by gschintgen** [comment]
> We should make a function `evalf` that also works for list/set/tuple.

I'm not sure what you mean. Isn't `EvalfMixin` supposed to be the tool to add `evalf` to a class? It's just that it seems not really possible to actually get properly evaluated results out of it. (#5683)

**Comment by oscarbenjamin** [comment]
I mean that there should be a function `evalf(obj)` as a counter-part to the method `obj.evalf()` as is the case for other things like `expand`, `simplify` etc. The function could handle native Python containers like list, set, tuple that can't have an `.evalf` method and then you can do:
```julia
In [3]: s = {sqrt(2), sqrt(3)}                                                                                                                 

In [4]: s                                                                                                                                      
Out[4]: {√2, √3}

In [5]: evalf(s)
Out[5]: {1.4142135623731, 1.73205080756888
```
This reminded me of #18181 but I realise now that this is very tangential point :)

**Comment by mhardcastle** [comment]
Well for what it's worth I agree with you @oscarbenjamin , that would be a useful convenience function. Possibly deserves a separate issue though (:

**Comment by mohitacecode** [comment]
should I open a new issue and submit a pr?
